test/local/mysql_tests.js

```diff
@@ -221,6 +221,109 @@ DB.connect(config)
       )
 
       test(
+        'db.countAccountsCreatedBefore(Date.now())',
+        function (t) {
+          t.plan(1)
+          return db.countAccountsCreatedBefore(Date.now())
+          .then(function(result) {
+            t.ok(result.count >= 0, 'returned non-negative count')
+          }, function() {
+            t.fail('should have succeeded')
+          })
+        }
+      )
+
+      test(
+        'db.countAccountsCreatedBefore(0)',
+        function (t) {
+          return db.countAccountsCreatedBefore(0)
+          .then(function(result) {
+            // Count should be zero if STRICT_ALL_TABLES and
+            // STRICT_TRANS_TABLES are not set in my.cnf
+            t.plan(1)
+            t.equal(result.count, 0)
+          }, function(err) {
+            // Query should fail if STRICT_ALL_TABLES or
+            // STRICT_TRANS_TABLES are set in my.cnf
```

Comment:
I only know about it accidentally tbh, after the first set of assertions I wrote failed unexpectedly.

Based on what I read [here](https://dev.mysql.com/doc/refman/5.6/en/sql-mode.html) it's a global strict mode setting for the server itself, which controls whether certain conditions should result in an error or an empty result set.

Because I didn't know how it would be set for other people or in deployed environments, I wrote the assertions to cover both cases. And then I felt uneasy about it so I added the comment. :confused: